server/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -94,6 +94,11 @@ public final class Script implements ToXContentObject, Writeable {
      */
     public static final String CONTENT_TYPE_OPTION = "content_type";
 
+    /**
+     * Compiler option to enable missing parameters detection.
+     */
+    public static final String DETECT_MISSING_PARAMS_OPTION = "detect_missing_params";
```

Comment:
Not sure this belongs here as it will only apply to Mustache.

Reply:
Now a constant of `MustacheScriptEngine`:

I wanted to avoid this initially since it creates a direct dependency to module:mustache for the ml module. But it is required to create this dependency to address this comment: https://github.com/elastic/elasticsearch/pull/102601#discussion_r1410624219